app-emulation/wine-staging/wine-staging-10.5.ebuild

```diff
@@ -3,7 +3,6 @@
 
 EAPI=8
 
-MULTILIB_COMPAT=( abi_x86_{32,64} )
 PYTHON_COMPAT=( python3_{10..13} )
 inherit autotools edo flag-o-matic multilib multilib-build optfeature
```

Comment:
Please:
1. drop multilib-build (keep multilib)
2. add "abi_x86_32 abi_x86_64" in IUSE
3. drop x32's package.mask

This will be closer to the old MULTILIB_COMPAT behaviour

Should work in theory, but possible that I'm overlooking something given haven't tested.

Reply:
multilib-build is the one that defines MULTILIB_USEDEP, do you want me to inline that part of the eclass?

Reply:
Ah right, there's that too... inlining is what I do in e.g. nvidia-drivers, but think should keep as-is for now. I don't want too many changes to keep up with until potential refactoring (pain to keep all the wine ebuilds in sync).

Sorry but please drop the idea and instead base/package.use.mask abi_x86_32 then (rather than package.mask).

Reply:
Still does not work unfortunately, pkgcheck complains about nonsolvable deps
```
NonsolvableDepsInDev: version 10.5: nonsolvable depset(bdepend) keyword(~amd64) dev profile (default/linux/amd64/23.0/x32): solutions: [ >=dev-util/mingw64-toolchain-10.0.0_p1-r2[abi_x86_32(-),abi_x86_64(-),abi_x86_x32(-)] ]
```

Reply:
Actually I'm now thinking that it's a good occasion to refactor and offload toolchain bits to a wine.eclass and sort this out properly with less code duplication across ebuilds.

Rather than make you change things around further, I'll look at incorporating what you have here into the eclass, and then report back so you can check if arm64 works with it.

Not sure how long it'll take me to get this ready though, shouldn't be too bad but depending on free time it could drag a few days.